chapters/sparse.rst

```diff
@@ -0,0 +1,261 @@
+
+When creating your own sparse matrix,
+we suggest you use `COO` as default matrix type.
+
+.. list-table:: Sparse Matrix Formats
+   :widths: 25 25 25 25 25
+   :header-rows: 1
+
+   * - ShortName
+     - FullName
+     - Scipy
+     - Tensorflow
+     - Pytorch
+   * - COO
+     - COOrdinate format
+     - Yes
+     - Yes
+     - Yes
+   * - BSR
+     - Block Sparse Row matrix
+     - Yes
+     - No
+     - No
+   * - CSC
+     - Compressed Sparse Column matrix
+     - Yes
+     - No
+     - No
+   * - CSR
+     - Compressed Sparse Row matrix
+     - Yes
+     - No
+     - No
+   * - DIA
+     - Sparse matrix with DIAgonal storage
+     - Yes
+     - No
+     - No
+   * - DOK
+     - Dictionary Of Keys based sparse matrix.
+     - Yes
+     - No
+     - No
+   * - LIL
+     - Row-based list of lists sparse matrix
+     - Yes
+     - No
+     - No
+
+
+Define your Jina Sparse Encoder
+---------------------------------
+
+After defining your Jina `Document` with Jina's Primitive Types,
+You need to use a sparse encoder to encode the `content` into a sparse representation,
+for instance, a Scipy `COO` matrix.
+We'll create a simple COO Encoder first
+
+.. highlight:: python
+.. code-block:: python
+
+    from scipy.sparse import coo_matrix
+    from jina.executors.encoders import BaseEncoder
+
+    class SimpleScipyCOOEncoder(BaseEncoder):
+
+        def encode(self, content: 'np.ndarray', *args, **kwargs) -> Any:
+            """Encode document content into `coo` format."""
+            return coo_matrix(content)
+
+Then we're able to make use of the `SimpleScipyCOOEncoder` defined above,
+inside the Jina Index and Search Flow.
+
+In Jina-Hub, we have created the `TFIDFTextEncoder <https://github.com/jina-ai/jina-hub/tree/master/encoders/nlp/TFIDFTextEncoder>`_ to encode text into sparse representation,
+the ``TFIDFTextEncoder`` is a wrapper on top of scikit-learn ``TfidfVectorizer`` to encode text data into a Scipy sparse matrix.
+
+
+You can create your own Encoder to encode your Document content into the expected format.
+
+Use a Jina Sparse Indexer
+--------------------------
+
+In Jina, we've created several Indexers to help you encode your Document content into sparse format.
+For instance, `PysparnnIndexer <https://github.com/jina-ai/jina-hub/tree/master/indexers/vector/PysparnnIndexer>`_
+is a library for fast similarity search of Sparse Scipy vectors.
+In contains an algorithm that can be used to perform fast approximate search with sparse inputs.
+Developed by Facebook AI Research.
+
+Build your Sparse Pipeline In Action
+--------------------------------------
+
+In this pipeline, we will make use of Jina's ``TFIDFTextEncoder`` together with ``PysparnnIndexer`` for encoding and indexing.
+
+As was mentioned before, ``TFIDFTextEncoder`` was created based on Scikit-learn,
+before using the Encoder, you need to ``fit`` the vectorizer with your training data.
+
+.. highlight:: python
```

Comment:
Add a sentence such as:

In this example, we use a simple corpus containing four sentences of text.